File: backends/ruby-hanami/app/usecase/staff/dto.rb
# frozen_string_literal: true
#
# スタッフユースケースの DTO を定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module UseCase
  module Staff
    # スタッフのロール更新時に渡す DTO です。
    UpdateRoleDto = Struct.new(:id, :role, :executor_id, :version, keyword_init: true)

    # スタッフ削除時に渡す DTO です。
    DestroyDto    = Struct.new(:id, :executor_id, :version, keyword_init: true)

    # スタッフ復元時に渡す DTO です。
    RestoreDto    = Struct.new(:id, :version, keyword_init: true)
  end
end

File: backends/go-echo/internal/usecase/staff/dto.go
// Package staff はスタッフユースケースの入出力データを定義します。
package staff

// UpdateRoleDto はスタッフロール変更の入力データです。
type UpdateRoleDto struct {
	// ID はスタッフID。
	ID uint
	// Role は変更後のロール。
	Role int
	// ExecutorID は操作を実行したスタッフID。
	ExecutorID uint
	// Version は楽観排他ロック用バージョン番号。
	Version int
}

// DestroyDto はスタッフ削除の入力データです。
type DestroyDto struct {
	// ID はスタッフID。
	ID uint
	// ExecutorID は操作を実行したスタッフID。
	ExecutorID uint
	// Version は楽観排他ロック用バージョン番号。
	Version int
}

// RestoreDto はスタッフ復元の入力データです。
type RestoreDto struct {
	// ID はスタッフID。
	ID uint
	// Version は楽観排他ロック用バージョン番号。
	Version int
}
